slate-social-media/src/Components/BookMarkedPost/Bookmarkedpost.js

```diff
@@ -1,31 +1,57 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { Link } from "react-router-dom";
+import { useLoginSignupContext } from "../Context/LoginSignupContext";
+import { useUserContext } from "../Context/UserContext";
```

Comment:
instead of importing context from every single files add one root-level file in particular folder and export it there then import from that file